### PR TITLE
List ssh keys

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Fetch sshkeys.
- Display the sshkeys in a table. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1506
- Add a component for toggling expanding rows and dedupe existing buttons. Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/190.

QA:
- In your MAAS add some ssh keys (/MAAS/account/prefs/).
- Visit /account/prefs/ssh-keys.
- Check that you get a list of ssh keys.

### CODE REVIEW

File: ui/src/app/preferences/reducers/sshkey/sshkey.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      case "FETCH_SSHKEY_START":
+        draft.loading = true;
+        break;
+      case "FETCH_SSHKEY_SUCCESS":
+        draft.loading = false;
+        draft.loaded = true;
+        draft.items = action.payload;
+        break;
+      default:
+        return draft;
```
Comment (squidsoup) [inline ui/src/app/preferences/reducers/sshkey/sshkey.js]: This should also reduce errors.

File: ui/src/app/preferences/views/SSHKeyList/SSHKeyList.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  beforeEach(() => {
+    state = {
+      sshkey: {
+        loading: false,
+        loaded: true,
+        items: [
+          {
+            id: 1,
+            key: "ssh-rsa aabb",
+            keysource: { protocol: "lp", auth_id: "koalaparty" }
```
Comment (squidsoup) [inline ui/src/app/preferences/views/SSHKeyList/SSHKeyList.test.js]: Koala parties must be pretty low key.

File: ui/src/app/preferences/views/SSHKeyList/SSHKeyList.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      groupId = `${protocol}/${auth_id}`;
+      id = auth_id;
+      source =
+        (protocol === "lp" && "Launchpad") ||
+        (protocol === "gh" && "GitHub") ||
+        protocol;
+    } else {
+      source = "Upload";
+      groupId = sshkey.id;
+    }
```
Comment (squidsoup) [inline ui/src/app/preferences/views/SSHKeyList/SSHKeyList.js]: Might be nice to have a few unittests for this logic, confirming that source is set correctly, or perhaps just another component test?

File: ui/src/app/preferences/reducers/sshkey/sshkey.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/preferences/reducers/sshkey/sshkey.js]: Done.

File: ui/src/app/preferences/views/SSHKeyList/SSHKeyList.test.js (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/preferences/views/SSHKeyList/SSHKeyList.test.js]: Yeah, it's mostly just ambient jams.

**Pull request merged**